django/contrib/postgres/apps.py

```diff
@@ -12,6 +13,10 @@ class PostgresConfig(AppConfig):
     verbose_name = _('PostgreSQL extensions')
 
     def ready(self):
+        # Closing any open connection to ensure register_hstore_handler is
+        # called for any new connection (#25454).
+        for conn in connections.all():
+            conn.close()
```

Comment:
If I remember correctly there was discussions about making sure `ready()` is always idempotent as it can be called multiple times (during testing when using `allowed_apps` or overriding `INSTALLED_APPS` if I remember correctly).

Now both alternatives aren't technically idempotent but closing all opened connections looks like it has more chance of having undesirable side effects then executing queries on already opened connections in a testing scenario given how Django's testing infrastructure uses transactions for example.